File: generate-ico.js
import pngToIco from 'png-to-ico';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const iconDir = path.join(__dirname, 'src-tauri/icons');
const pngPath = path.join(iconDir, 'icon.png');
const icoPath = path.join(iconDir, 'icon.ico');

async function convertToIco() {
  try {
    await pngToIco(pngPath, icoPath);
    console.log('✓ Created icon.ico');
  } catch (error) {
    console.error('Error creating ICO:', error.message);
  }
}

convertToIco();

File: generate-ico.py
import os
from PIL import Image

icon_dir = 'src-tauri/icons'
icon_path = os.path.join(icon_dir, 'icon.png')
output_ico = os.path.join(icon_dir, 'icon.ico')

# Create ICO file from PNG
img = Image.open(icon_path)
img.save(output_ico, format='ICO')

print(f'✓ Created {output_ico}')
